### PR TITLE
Refactors status code handling in Airbyte client

### LINKED ISSUE #17: Randomly cannot unpack error from trigger_manual_sync_connection
Hi,

I'm reporting a weird behavior I'm facing while trying to trigger prefect-airbyte task. Most of the time it works well but sometimes I got this error : 
```
job_id, job_created_at = airbyte.trigger_manual_sync_connection(
TypeError: cannot unpack non-iterable NoneType object
``` 

It would mean that this function returns a None object but not specific errors which makes the debugging pretty tricky.

On the airbyte side, the prefect task has created 2 jobs: one failing and one successful.

Full log https://pastebin.com/4xZBJAYd

### PR DESCRIPTION
**Pull request by desertaxle**
## Summary

Refactors the status code handling in the Airbyte client to handle all non 2xx status codes. Previously, status codes that did not match the preconfigured checks would fall through and cause unexpected errors. Now, we will raise on all non 2xx status codes and handle the exceptions in specific cases.

## Relevant Issue(s)

Closes #17 

## Checklist
- [x] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-airbyte/blob/main/CHANGELOG.md)

### CODE REVIEW

File: prefect_airbyte/client.py [hunk 1]
```diff
@@ -205,13 +202,13 @@ async def get_job_status(self, job_id: str) -> str:
...
-                job_updated_at = response.json()["job"]["updatedAt"]
-                return job_status, job_created_at, job_updated_at
-            elif response.status_code == 404:
-                self.logger.error(f"Job {job_id} not found...")
-                raise err.JobNotFoundException(f"Job {job_id} not found...")
+            response.raise_for_status()
+
+            job_status = response.json()["job"]["status"]
+            job_created_at = response.json()["job"]["createdAt"]
+            job_updated_at = response.json()["job"]["updatedAt"]
```
Comment (ahuang11) [inline prefect_airbyte/client.py:209]:
```suggestion
            job = response.json()["job"]
            job_status = job["status"]
            job_created_at = job["createdAt"]
            job_updated_at = job["updatedAt"]
```

File: prefect_airbyte/client.py [hunk 2]
```diff
@@ -205,13 +202,14 @@ async def get_job_status(self, job_id: str) -> str:
...
-            elif response.status_code == 404:
-                self.logger.error(f"Job {job_id} not found...")
-                raise err.JobNotFoundException(f"Job {job_id} not found...")
+            response.raise_for_status()
+
+            job = response.json()["job"]
+            job_status = job["status"]
+            job_created_at = job["createdAt"]
+            job_updated_at = job["updatedAt"]
+            return job_status, job_created_at, job_updated_at
```
Comment (ahuang11) [inline prefect_airbyte/client.py:211]: I think the `Returns` section should be more explicit on what it return; and I don't think it's a byte array?

File: prefect_airbyte/client.py [hunk 3]
```diff
@@ -172,25 +174,20 @@ async def trigger_manual_sync_connection(
...
-            elif response.status_code == 404:
-                # connection_id not found
-                self.logger.warning(
-                    f"Connection {connection_id} not found, please double "
-                    f"check the connection_id ..."
-                )
+            response.raise_for_status()
+            job_id = response.json()["job"]["id"]
+            job_created_at = response.json()["job"]["createdAt"]
+            return job_id, job_created_at
```
Comment (ahuang11) [inline prefect_airbyte/client.py:180]:
```suggestion
            response.raise_for_status()
      
            job = response.json()["job"]
            job_id = job["id"]
            job_created_at = job["createdAt"]
            return job_id, job_created_at
```

I think the `Returns:` docstring section needs reformatting

**Review by ahuang11 — CHANGES_REQUESTED** [review]

File: prefect_airbyte/client.py (hunk 2 above)
Comment (desertaxle) [reply prefect_airbyte/client.py:198]: Addressed in [fcab2c8](https://github.com/PrefectHQ/prefect-airbyte/pull/20/commits/fcab2c88241fa672d36e4bb9e996c3436258b62c)

**Review by ahuang11 — APPROVED** [review]

**Pull request merged**